pkg/agent/config/config.go

```diff
@@ -81,7 +81,7 @@ func getNodeNamedCrt(nodeName, nodeIP, nodePasswordFile string) HTTPRequester {
 			return nil, err
 		}
 		req.Header.Set(version.Program+"-Node-Password", nodePassword)
-		req.Header.Set(version.Program+"-Node-IP", nodeIP)
+		req.Header.Set(version.Program+"-Node-IP", nodeIPs[0])
```

Comment:
So there is a lot of fallout in signatures to make them start taking `[]string`, then we throw that all away by taking the 0th element. 

An alternative that might be a bit cleaner is to actually not change the signature (b/c to me it hides the fact that we're only using the 0 position) and push that responsibility higher in the call chain. So somewhere in the `get` method that calls `getServingCert` and `getNodeNamedHostFile`